### DISCUSSION TITLE
Firewall issue?

### DISCUSSION THREAD

**Discussion by airocjn** [GerryDazoo/Slinger#97] [Q&A]
Hello,

I'm close to loading my stream from VLC player on Windows, however, when it makes a connection, I see the video, but it quickly ends and I'm seeing this error:  [WinError 10053] An established connection was aborted by the software in your host machine. 

I'm not sure where to go from here. I already allowed slingbox_server.exe in Windows Defender Firewall and I port forwarded 5201 for the Slingbox connected on my loal network. Is there another setting I'm missing?

**Comment by donscreen** [comment]
I had the same errors when I had multiple connection windows open when testing.  Make sure they are closed before running a new connection.   I did get mine working with a Slingbox 500 and M2 model.   I  have 2 Tivo Premieres hooked to the Slingbox 500 and a Tivo Premiere XL hooked to the Slingbox M2.   I used a separate config.ini file for each Tivo.   I created a server shortcut with the Windows exe file and each of the 3 configs on the command line (using shortcut properties).  I used ports 8082, 8084 and 8086.
I only need to run the one server exe file shortcut to be able to access both slingboxes and all 3 Tivos.  MK Player works well on the Firestick and I use VLC on my mac and Windows 11.  Thank you Gerry!

**Reply by airocjn** [reply]
Thanks for this info @donscreen. I only have one connection window open to test, so I don't think that's the problem. Any other ideas?

**Reply by donscreen** [reply]
Sorry....that was my solution to the issue.  Maybe Gerry has a better idea of how to fix this...

**Reply by GerryDazoo** [reply]
If the steam is starting up then It's  not a firewall issue. Please try a different media player app to see if you have the same issue.

**Reply by f520k510** [reply]
> Thanks for this info @donscreen. I only have one connection window open to test, so I don't think that's the problem. Any other ideas?

@airocjn have you figured it out?  I am having the same win error 10053 issue.  The video stream would show but soon stop.  I added exception to my mcafee total protection firewall but in no avail.

Next I will try re-installing vlc and changing a new video player.

**Reply by airocjn** [reply]
Yes, I was able to use the PlayerXtreme app to access the stream. But I also reinstalled VLC player to make sure I was using the latest version and I’m using the newest version of Slinger that Gerry released, and now VLC works! I hope it works for you too. 

Now I’m trying to access the stream outside my local network, but I’m not having any luck even after following the instructions. I’m guessing it’s an issue with either firewall, ISP, or router setup.

**Reply by f520k510** [reply]
Did you enter your slingbox's wan ip instead of its internal ip in the config.ini?
That works for me.

**Reply by jchan81** [reply]
I'm having the exact same issue, I would get 0.5 second of video / audio and it would crash out.  If I keep hammering the connection it would occasionally work but if I change the channel on my STB, I lose connection to it.  I would need to then restart the service.  I have a Cisco 8642HD connected via HDMI.

**Reply by GerryDazoo** [reply]
Make sure you have the proper SBType configured. Please post all the server output and your config.ini. What type of slingbox do you have?

**Reply by jchan81** [reply]
I have a 500
**Config.ini:**

[SLINGBOX]
sbtype="500"
;sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=xxxxx
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your 
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. Also removes requirement for the netifaces module  
ipaddress=192.168.1.110
port=5201
; Repace with your finderid, If and only if you need remote access and don't  
;have a static ip and you plan on using my service to access you server remotely. 
; Please read the release notes. Most people don't need this.
;finderid=xxxxxx

;If you've got more than one slingbox set name. THe server will use this when generating 
; logs to make it easier to see what's going on. 
;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
VideoBandwidth=2000

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, 1=composite 2=svideo 3=component
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't set this the code will use the last configired input
; I recommend setting this value. If not set correctly you'll often only see a black screen. 
VideoSource = 2

[SERVER]
; local port number for the server to listen on for connections
port=8080
;maxstreams=10
;enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
;include=remote.txt 


**Output**

Version : 3.08e Running on Windows-10-10.0.19044-SP0 pid= 3740
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.110', 5201)
('192.168.1.110', 5201) OK
8080 Using slingbox at  ('192.168.1.110', 5201)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
11/16/2022, 16:54:55.202   Streaming request from ('192.168.1.163', 54320)

Slinginfo  "500" 12 30 192.168.1.110 5201 3072 10
8080 Starting Stream for  192.168.1.163:54320
Connecting... ('192.168.1.110', 5201) Control
8080 Selecting Video Source 2
Keycodes= [1, 4, 5, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 21, 22, 24, 25, 26, 27, 28, 29, 31, 32, 33, 34, 35, 36, 38, 39, 40, 41, 42, 43, 44, 45, 46, 56, 57, 59, 60, 62, 63, 64, 65, 66, 67, 68, 69, 70, 74, 75, 76, 77]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.110', 5201) Stream
8080 Stream started at 11/16/2022, 16:54:57.60  744 0
11/16/2022, 16:54:59.874  8080 Stream Terminated for  192.168.1.163:54320 [WinError 10053] An established connection was aborted by the software in your host machine
8080 Shutting down connections
8080 Logging Out
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
JCHAN81. Your problem with the connection dropping on channel channel is an issue with VLC. SMPlayer and ffPlayer do handle this correctly.

**Reply by jchan81** [reply]
Any suggestions for iOS?
